### PR TITLE
Added client side camera movement

### PR DESCRIPTION
**Pull request by VicFrank**
Panorama camera movement, so beautiful, so pure.

### CODE REVIEW

File: common/events.d.ts [hunk 1]
```diff
@@ -6,8 +6,17 @@ interface SectionStartedEvent {
     section: SectionName;
 }
 
+interface MoveCameraEvent {
+    cameraTargetX?: number;
+    cameraTargetY?: number;
+    cameraTargetZ?: number;
+    unitTargetEntIndex?: EntityIndex;
+    lerp: number;
```
Comment (Perryvw) [inline common/events.d.ts:14]: What is lerp? Duration?

**Review by Perryvw — APPROVED** [review]

File: common/events.d.ts (hunk 1 above)
Comment (Shushishtok) [reply common/events.d.ts:14]: Should probably be renamed "duration" and used as a duration - the time it takes for the camera to pan from the unit to the target.

File: content/panorama/scripts/custom_game/move-camera.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    GameEvents.Subscribe("move_camera", (event) => {
+        let cameraTarget: [number, number, number] = [0, 0, 0];
+        if (event.unitTargetEntIndex) {
+            cameraTarget = Entities.GetAbsOrigin(event.unitTargetEntIndex);
+        } else if (event.cameraTargetX && event.cameraTargetY && event.cameraTargetZ) {
+            cameraTarget[0] = event.cameraTargetX;
+            cameraTarget[1] = event.cameraTargetY;
+            cameraTarget[2] = event.cameraTargetZ;
+        }
+        GameUI.SetCameraTargetPosition(cameraTarget, event.lerp);
```
Comment (Shushishtok) [inline content/panorama/scripts/custom_game/move-camera.ts:11]: Does that lock the camera of the player to the frame? What if the player is trying to scroll the camera?
If it does lock the camera, does it releases it afterwards?

File: game/scripts/vscripts/TutorialGraph/Steps.ts [hunk 3]
```diff
@@ -115,6 +115,48 @@ export const setCameraTarget = (target: CBaseEntity | undefined) => {
...
+    })
+}
+
+/**
+ * Moves the camera to a position, with lerp
+ * @param position Point to move the camera to.
+ * @param lerp Speed at which the camera moves
+ */
+export const moveCameraToPosition = (position: Vector, lerp: number) => {
+    let playerIds = findAllPlayersID();
```
Comment (Shushishtok) [inline game/scripts/vscripts/TutorialGraph/Steps.ts:144]: I think you can just use `findRealPlayerID()` instead to get the player's ID. There's only one real player.

**Pull request merged**